packages/design-system/src/SegmentedControl/SegmentedControl.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import clsx from "classnames";
+
+import { SegmentedControlProps } from "./SegmentedControl.types";
+import {
+  StyledSegmentedControl,
+  ControlOption,
+  ControlContainer,
+} from "./SegmentedControl.styles";
+import { Icon } from "Icon";
+import { Text } from "Text";
+import {
+  SegmentedControlClassName,
+  SegmentedControlOptionClassName,
+  SegmentedControlOptionContainerClassName,
+} from "./SegmentedControl.constants";
+
+function SegmentedControl(props: SegmentedControlProps) {
+  const { className, defaultValue, onChange, options, ...rest } = props;
+  const [selectedValue, setSelectedValue] = useState(defaultValue);
+
+  const handleOnChange = (value: string) => {
+    setSelectedValue(value);
+    onChange && onChange(value);
+  };
+
+  return (
+    <StyledSegmentedControl
+      className={clsx(SegmentedControlClassName, className)}
+      {...rest}
+    >
+      {options.map((option) => {
+        const {
+          endIcon,
+          isDisabled: disableControl,
+          label,
+          startIcon,
+          value,
+        } = option;
+
+        return (
+          <ControlContainer
+            className={SegmentedControlOptionContainerClassName}
+            data-disabled={disableControl}
+            data-selected={selectedValue === value}
+            key={value}
+            onClick={() => !disableControl && handleOnChange(value)}
+          >
+            <ControlOption
+              className={SegmentedControlOptionClassName}
+              data-selected={selectedValue === value}
+            >
+              {/* if icon name is passed */}
```

Comment:
😳 Is this something we will have to do every where we use icon? Can we find a way to handle this within icon itself?

Reply:
Yes for now, you can give it a try